src/syntax_mapping.rs

```diff
@@ -122,29 +122,44 @@ impl<'a> SyntaxMapping<'a> {
             .insert("*.hook", MappingTarget::MapTo("INI"))
             .unwrap();
 
-        if let Some(xdg_config_home) = std::env::var_os("XDG_CONFIG_HOME") {
-            let git_config_path = Path::new(&xdg_config_home).join("git");
+        // global git config files
+        // `$XDG_CONFIG_HOME/git`, or `$HOME/.config/git` if `$XDG_CONFIG_HOME` is not set or empty
+        // see https://git-scm.com/docs/git-config#FILES
+        // we cover both cases regardless of the state of `$XDG_CONFIG_HOME`
+        if let Some(xdg_config_home) =
+            std::env::var_os("XDG_CONFIG_HOME").and_then(|val| (!val.is_empty()).then(|| val))
+        {
+            insert_git_config_global(&mut mapping, &xdg_config_home);
+        }
+        if let Some(default_config_home) =
+            std::env::var_os("HOME").map(|home| Path::new(&home).join(".config"))
+        {
+            insert_git_config_global(&mut mapping, &default_config_home);
+        }
+
+        fn insert_git_config_global(mapping: &mut SyntaxMapping, config_home: impl AsRef<Path>) {
+            let git_config_path = config_home.as_ref().join("git");
 
             mapping
                 .insert(
                     &git_config_path.join("config").to_string_lossy(),
                     MappingTarget::MapTo("Git Config"),
                 )
-                .ok();
+                .unwrap();
```

Comment:
Why change to unwrap here?

Reply:
I believe `Result::ok` here is semantically inaccurate. If I understand correctly, `SyntaxMapping::insert` is infallible in this specific context, and therefore `unwrap` (or `expect`) should be used. If `SyntaxMapping::insert` is fallible here then I stand corrected.

In the original PR (#1195), @sharkdp stated "`XDG_CONFIG_HOME` could contain invalid UTF8" as the reason for removing `unwrap`, but that applies to `git_config_path.join("config").into_os_string().to_str().unwrap()`, not to this `insert` call. So I suspect they might have made this inaccuracy in a rush.

Reply:
I appreciate you wanting get the code semantically accurate. I tried to convince myself of that insert() can never fail, but it was not obvious to me. While it wasn't obvious that it can fail either, I would prefer if we could play it safe and keep `ok()` for now.

Reply:
Personally I prefer to fail fast if `insert` fails here, hence `unwrap`. But I'm happy to respect your preference if you insist.

Reply:
Failing fast is good for developers, but not for users. Users will prefer to be shown the file they requested bat to show, albeit without syntax highlighting, over bat crashing on them. So I would say I insist on reverting back to `.ok()`.

Reply:
Sure no problem.